CMakeLists.txt

```diff
@@ -472,6 +470,19 @@ if (BUILD_TOOLS)
     INSTALL
   )
 
+if (NOT WIN32)
+  set(LDL_LIB "-ldl")
+else()
+  set(LDL_LIB "")
+endif()
+
+  wabt_executable(
+    NAME wasm2c-runner
+    SOURCES wasm2c/wasm-rt-runner.c
```

Comment:
This is a pattern that is common in many sandboxing toolchains. Lucet is probably the easiest to look at https://github.com/bytecodealliance/lucet/blob/main/lucet-wasi/src/main.rs

The goal of the runner is to give developers an easy way to test simple applications without changing their code. The runner uses a default Wasm configuration (instantiates a heap and the runtime etc) and would execute any wasm2c compiled module that includes a main. The expected workflow is that developers can simply compile their existing application to a wasm module, generate C with wasm2c, compile the result to a dll and run it with no modification.

Reply:
I guess the separation of runtime from compiled module is the key benefit here..   since it allows the compilation of the module and runtime to be separated.   But it seems that benefit is largely proportional to size of the runtime itself and how likely it is that you would want to iterate on the runtime without recompiling the wasm2c code.

In practice, have you found this useful with wasm2c?

Reply:
Yup! In particular, I was compiling SPEC from wasm at some point and this was a useful way to not have to compile wasm2c repeatedly for each of the SPEC benchmarks. You are absolutely right in that all this does is allow users to avoid compiling wasm2c multiple times. 

One additional (minor benefit) is that  having a runner is nice as it simplifies tutorials.

One last datapoint is that other Wasm compilers like Lucet really needed a runner as recompiling Lucet (which is written in rust) takes a while. Wasm2c is small enough that we can make either case here. 

Given this, I think a runner would be nice to have but not a strict necessity.